tests/checks/symlinks-not-overwritten.fish

```diff
@@ -0,0 +1,45 @@
+#RUN: %fish -C 'set -g fish %fish' %s
+begin
+    # fish_variables
+    set -gx XDG_CONFIG_HOME (mktemp -d)
+    mkdir -p $XDG_CONFIG_HOME/fish
+    set -l target_file $XDG_CONFIG_HOME/fish/target_fish_variables
+    set -l fish_variables $XDG_CONFIG_HOME/fish/fish_variables
+
+    echo >$target_file
+    ln -sf $target_file $fish_variables
+    $fish -c 'set -U variable value'
+
+    if not test -L $fish_variables
+        echo fish_variables has been overwritten
+    else if test (wc -c < $target_file) -le 3
```

Comment:
this is a bit tricky - I wonder if there is an easier way to write this.

Maybe `grep -v . $target_file`, or the backup file trick below?

Reply:
I'll just do the backup trick. I guess `grep` would work, but I'm vaguely afraid of `echo` writing `\r\n` on Windows and `grep` not understanding it. Or something.